### PR TITLE
[release-0.25] Fix channel template spec type for Sequence channel template field

### PR DESCRIPTION
**Pull request by pierDipi**
Cherry-pick of: https://github.com/knative/eventing/pull/5979/files

### CODE REVIEW

**Comment by pierDipi** [comment]
```
rhcloud.com:6443/apis/config.openshift.io/v1/clusterversions/version": dial tcp: lookup api.ci-op-zg3wtr8w-0301d.origin-ci-int-aws.dev.rhcloud.com on 172.30.0.10:53: no such host
ClusterID: 
ClusterVersion: Installing "" for <unknown>: <unknown>
error getting cluster operators: Get "https://api.ci-op-zg3wtr8w-0301d.origin-ci-int-aws.dev.rhcloud.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp: lookup api.ci-op-zg3wtr8w-0301d.origin-ci-int-aws.dev.rhcloud.com on 172.30.0.10:53: no such host
ClusterOperators:
	clusteroperators are missing
```
/test 47-e2e-aws-ocp-47

**Comment by pierDipi** [comment]
```
error: gather did not start for pod must-gather-b672s: unable to pull image: ImagePullBackOff: Back-off pulling image "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a4dc776b581ba6c4b2d1a5a08ccd09598177e2934e9159d0eac4067b1bdcce01"
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-12-07T17:20:17Z"}
error: failed to execute wrapped command: exit status 1
```
/test 49-reconciler-aws-ocp-49

File: vendor/knative.dev/reconciler-test/pkg/eventshub/103-pod.yaml [hunk 1]
```diff
@@ -24,7 +24,7 @@ spec:
   restartPolicy: "Never"
   containers:
     - name: eventshub
-      image: registry.ci.openshift.org/openshift/knative-nightly:knative-eventing-test-eventshub
+      image: {{ .image }}
```
Comment (matzew) [inline vendor/knative.dev/reconciler-test/pkg/eventshub/103-pod.yaml:27]: STOP! 

/hold

Comment (matzew) [reply vendor/knative.dev/reconciler-test/pkg/eventshub/103-pod.yaml:27]: we must wait on: https://github.com/openshift/knative-eventing/pull/1545

File: vendor/knative.dev/reconciler-test/pkg/images/ko.go [hunk 2]
```diff
@@ -16,7 +16,20 @@ limitations under the License.
...
 // Use ko to publish the image.
 func KoPublish(path string) (string, error) {
-	return "", nil
+	platform := os.Getenv("PLATFORM")
+	if len(platform) > 0 {
+		platform = " --platform=" + platform
+	}
+	out, err := runCmd(fmt.Sprintf("ko publish%s -B %s", platform, path))
+	if err != nil {
+		return "", err
```
Comment (matzew) [inline vendor/knative.dev/reconciler-test/pkg/images/ko.go:32]: no - we patched this tool, to not have `ko`

**Comment by matzew** [comment]
Do we really want this for the already released 1.19? 

If so: we should udpate this PR with: 
* cherry pick of _just_ this `sha`: https://github.com/knative/eventing/commit/6a0c6d47de2e10a2478a003f2a2f13d93f5387c4
* update the generated manifest

Do a PR against the serverless operator 1.19 branch 

My take is: we already have shipped the bug... and can point customers to next version. It's not a regression, it's been this for a longer... 

Given all the work above, and that 1.20 will come out "soon" - I am not sure it is really worth it for doing this of the 1.19 / 0.25 branches 

/cc @pierDipi @devguyio @lberk

**Comment by devguyio** [comment]
> Do we really want this for the already released 1.19?
> 
> If so: we should udpate this PR with:
> 
> * cherry pick of _just_ this `sha`: [knative@6a0c6d4](https://github.com/knative/eventing/commit/6a0c6d47de2e10a2478a003f2a2f13d93f5387c4)
> * update the generated manifest
> 
> Do a PR against the serverless operator 1.19 branch (like [openshift-knative/serverless-operator#1334](https://github.com/openshift-knative/serverless-operator/pull/1334))
> 
> My take is: we already have shipped the bug... and can point customers to next version. It's not a regression, it's been this for a longer...
> 
> Given all the work above, and that 1.20 will come out "soon" - I am not sure it is really worth it for doing this of the 1.19 / 0.25 branches
> 
> /cc @pierDipi @devguyio @lberk

I'm fine with skipping 1.19/0.25 given the proximity of 1.20/0.26

**Comment by pierDipi** [comment]
I'm ok with skipping this, closing ...

/close

**Pull request closed without merging**